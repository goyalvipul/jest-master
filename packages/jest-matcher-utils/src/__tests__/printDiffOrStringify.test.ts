/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {EXPECTED_COLOR, INVERTED_COLOR, printDiffOrStringify} from '../index';

describe('printDiffOrStringify', () => {
  const testDiffOrStringify = (expected: string, received: string): string =>
    printDiffOrStringify(expected, received, 'Expected', 'Received', true);

  test('expected is empty and received is single line', () => {
    const expected = '';
    const received = 'single line';
    expect(testDiffOrStringify(expected, received)).toMatchSnapshot();
  });

  test('expected is multi line and received is empty', () => {
    const expected = 'multi\nline';
    const received = '';
    expect(testDiffOrStringify(expected, received)).toMatchSnapshot();
  });

  test('expected and received are single line with multiple changes', () => {
    const expected = 'delete common expected common prev';
    const received = 'insert common received common next';
    expect(testDiffOrStringify(expected, received)).toMatchSnapshot();
  });

  test('expected and received are multi line with trailing spaces', () => {
    const expected = 'delete \ncommon expected common\nprev ';
    const received = 'insert \ncommon received common\nnext ';
    expect(testDiffOrStringify(expected, received)).toMatchSnapshot();
  });

  test('received is multiline longer than max', () => {
    const expected = 'multi\nline';
    const received = 'multi' + '\n123456789'.repeat(2000); // 5 + 20K chars

    const test = testDiffOrStringify(expected, received);

    // It is a generic line diff:
    expect(test).toContain(EXPECTED_COLOR('- line'));

    // It is not a specific substring diff
    expect(test).not.toContain(EXPECTED_COLOR('- ' + INVERTED_COLOR('line')));
  });
});
