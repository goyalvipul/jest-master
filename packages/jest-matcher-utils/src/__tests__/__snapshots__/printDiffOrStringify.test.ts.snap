// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printDiffOrStringify expected and received are multi line with trailing spaces 1`] = `
"<green>- Expected</>
<red>+ Received</>

<green>- <inverse>delete</>路</>
<red>+ <inverse>insert</>路</>
<green>- common <inverse>expect</>ed common</>
<red>+ common <inverse>receiv</>ed common</>
<green>- <inverse>prev</>路</>
<red>+ <inverse>next</>路</>"
`;

exports[`printDiffOrStringify expected and received are single line with multiple changes 1`] = `
"Expected: <green>\\"<inverse>delete</> common <inverse>expect</>ed common <inverse>prev</>\\"</>
Received: <red>\\"<inverse>insert</> common <inverse>receiv</>ed common <inverse>next</>\\"</>"
`;

exports[`printDiffOrStringify expected is empty and received is single line 1`] = `
"Expected: <green>\\"\\"</>
Received: <red>\\"single line\\"</>"
`;

exports[`printDiffOrStringify expected is multi line and received is empty 1`] = `
"Expected: <green>\\"multi</>
<green>line\\"</>
Received: <red>\\"\\"</>"
`;
